app/main.py

```diff
@@ -1,3 +1,45 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other):
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            return NotImplemented
+        return self
+
+    def __mul__(self, num: int):
+        return Distance(self.km * num)
+
+    def __truediv__(self, other: int):
+        return Distance(round(self.km / other, 2))
```

Comment:
The `__truediv__` method should include a check to prevent division by zero, which would raise a `ZeroDivisionError`. Consider adding a condition to handle this case.